### PR TITLE
Fixes #29851 - Add DSL docs for NIC

### PR DESCRIPTION
**Pull request by ofedoren**
For the reviewer: I'm not quite sure what should be in the description for `link` property. Any advices?

### CODE REVIEW

File: app/models/nic/base.rb [hunk 1]
```diff
@@ -77,8 +77,45 @@ class Base < ApplicationRecord
...
+      property :provision, one_of: [true, false], desc: 'Returns true if this controller is used for provisioning, false otherwise'
+      property :alias?, one_of: [true, false], desc: 'Returns true if this controller is used as an alias, false otherwise'
+      property :inheriting_mac, String, desc: 'Returns inherited MAC address of the controller this controller is alias for'
+      property :children_mac_addresses, Array, desc: 'Returns MAC addresses of attached devices'
+      property :nic_delay, Integer, desc: 'Returns the delay in seconds for network activity during install'
+      property :fqdn, String, desc: 'Returns FQDN for this device'
+      property :shortname, String, desc: 'Returns the controller\'s name without its domain'
+      property :type, String, desc: 'Returns type of this controller, e.g. Nic::Managed'
+      property :vlanid, String, desc: 'Returns VLAN ID of the subnet this device is associated with'
+      property :managed?, one_of: [true, false], desc: 'Returns true if the type is Nic::Managed, false otherwise'
```
Comment (shiramax) [inline app/models/nic/base.rb]: 'Returns True if external services such as DNS, DHCP and TFTP are configured for this interface'

File: app/models/nic/base.rb [hunk 2]
```diff
@@ -77,8 +77,45 @@ class Base < ApplicationRecord
...
+      property :bond?, one_of: [true, false], desc: 'Returns true if the type is Nic::Bond, false otherwise'
```
Comment (shiramax) [inline app/models/nic/base.rb]: 'Returns true if the type of the interface is Bond, false otherwise'

File: app/models/nic/base.rb [hunk 3]
```diff
@@ -77,8 +77,45 @@ class Base < ApplicationRecord
...
+      property :bridge?, one_of: [true, false], desc: 'Returns true if the type is Nic::Bridge, false otherwise'
```
Comment (shiramax) [inline app/models/nic/base.rb]: 'Returns true if the type of the interface is Bridge, false otherwise'

File: app/models/nic/base.rb [hunk 4]
```diff
@@ -77,8 +77,45 @@ class Base < ApplicationRecord
...
+      property :bmc?, one_of: [true, false], desc: 'Returns true if the type is Nic::BMC, false otherwise'
```
Comment (shiramax) [inline app/models/nic/base.rb]: the same..

File: app/models/nic/base.rb [hunk 5]
```diff
@@ -77,8 +77,45 @@ class Base < ApplicationRecord
...
+      property :link, one_of: [true, false], desc: 'TODO: not sure'
```
Comment (shiramax) [inline app/models/nic/base.rb]: I think: 
'Returns true if the Interface is up"

**Review by shiramax — CHANGES_REQUESTED** [review]

**Comment by ofedoren** [comment]
Thanks, @shiramax! Updated.

**Review by shiramax — APPROVED** [review]

**Pull request merged**